proposals/manifest_trial_tokens.md

```diff
@@ -0,0 +1,175 @@
+   from Manifest Version 1 to Manifest Version 2, and now from Manifest Version
+   2 to Manifest Version 3). This is taxing on developers since developers may
+   have to maintain code paths for both API behaviors for the entire duration of
+   migration.
+
+The new manifest key `trial_tokens` can enable per-change and per-extension
+opt-in leading to speedup in API iteration.
+
+See also [Chromium source code](https://github.com/chromium/chromium/blob/main/chrome/common/extensions/api/_api_features.json)
+
+### Known Consumers
+
+None at the moment.
+
+## Specification
+
+### Schema
+
+This is a manifest change with no programmable API.
+
+This proposal extends manifest with optional `trial_tokens` member which
+contains an array of strings which represent the stringified trial tokens.
+
+Parsed and recognized tokens should be exposed via existing
+[`runtime.getManifest()`](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/runtime/getManifest)
+API in `origin_trials`. Invalid or unrecognized tokens should be excluded
+from the returned manifest. This will enable extensions to detect recognized
+tokens and adjust accordingly.
+
+### Behavior
+
+#### Manifest parsing
+
+Upon manifest parsing, browser should parse `trial_tokens` like so:
+ - `trial_tokens` key must never cause installation errors, only benign
+   warnings
+ - browsers can limit the total number of parsed and/or accepted tokens
+ - browsers can limit the length of each token
+ - browsers can limit the the total length of all parsed and/or accepted tokens
+ - browsers can validate token structure, content, and signature at parse time;
+   in particular, tokens should not contain conrol characters like new lines
+   and charaxter returns which could affect HTTP header serealization and
+   parsing
+
+#### Manifest parsing algorithm
+ 1. if `trial_tokens` is not present, return
+ 2. if `trial_tokens` is not an array or if `trial_tokens` is an empty array,
+    (optionally) log a benign warning and return
+ 3. for every element `token` of `trial_tokens`:
+   3.1. (optionally) if the maximum number of accepted tokens is reached,
+        skip all other tokens
+   3.2. if `token` is not a string, skip it and (optionally) log a benign
+        warning
+   3.2. if `token` is an empty string, skip it and (optionally) log a benign
+        warning
+   3.3. (optionally) if `token` is too long to be parsed, skip it and
+        (optionally) log a benign warning
+   3.4. if token has been encountered before, skip it and (optionally) log
+        a benign warning
+   3.5. (optionally) attempt to parse the token and validate it. If validation
+        fails, skip the token and (optionally) log a benign warning.
+        Note: browsers may disregard token expiration errors, if they are not
+        sure about the corrctness of system timestamp at the time of token
+        validation
+   3.6. append token to the collection of accepted tokens
```

Comment:
GitHub is not formatting these as sub-items for some reason.

Reply:
Thanks, I fixed it.